detectable/src/main/java/com/synopsys/integration/detectable/detectables/yarn/YarnTransformer.java

```diff
@@ -91,6 +109,54 @@ private void addYarnLockDependenciesToGraph(YarnLockResult yarnLockResult, LazyE
             }
         }
     }
+    
+    private DependencyGraph buildGraphForProjectOrWorkspace(
+            LazyBuilderMissingExternalIdHandler lazyBuilderHandler,
+            ExternalIdDependencyGraphBuilder graphBuilder,
+            YarnLockResult yarnLockResult
+    ) throws MissingExternalIdException {
```

Comment:
This method seems to have a lot of nesting. Might make sense to break into smaller methods, I'm guessing Sonar might complain if you don't, or some comments at the top of the for loops might help in understanding.

Reply:
Done, thanks!